src/promptflow/promptflow/_sdk/_pf_client.py

```diff
@@ -37,6 +37,9 @@ def __init__(self, **kwargs):
         self._connections = None
         self._flows = FlowOperations(client=self)
         self._tools = ToolOperations()
+        # add user agent from kwargs if any
+        if isinstance(kwargs.get("user_agent"), str):
```

Comment:
This way actually looks strange. Because it's changing a global config in a class init function.

Reply:
Offline synced, according to currently design, setting this will indeed affect current session.